libs/api_common/map_descriptors.cpp

```diff
@@ -72,12 +72,30 @@ get_all_map_handles()
     return handles;
 }
 
+std::vector<map_cache_t>
+get_all_map_descriptors()
+{
+    std::vector<map_cache_t> descriptors;
+    size_t size = _map_file_descriptors.size();
+    for (size_t i = 0; i < size; i++) {
+        descriptors.push_back(_map_file_descriptors[i]);
+    }
+
+    return descriptors;
+}
+
 void
 cache_map_file_descriptor(uint32_t type, uint32_t key_size, uint32_t value_size, int fd)
 {
     _map_file_descriptors.push_back({(uintptr_t)fd, {fd, type, key_size, value_size, 0}});
 }
 
+void
+cache_map_file_descriptor_with_handle(uint32_t type, uint32_t key_size, uint32_t value_size, int fd, uintptr_t handle)
+{
+    _map_file_descriptors.push_back({handle, {fd, type, key_size, value_size, 0}});
```

Comment:
I think emplace_back would be more efficient than push_back here.  Any time you call a constructor + push_back, it should instead call emplace_back.   The difference is that emplace_back avoids a copy after construction whereas push_back does not.

Reply:
changed to emplace_back